index_mh_iter.go

```diff
@@ -1,22 +1,22 @@
-package suppliers
+package provider
 
 import (
 	"context"
 	"io"
 
-	provider "github.com/filecoin-project/indexer-reference-provider"
 	"github.com/ipld/go-car/v2/index"
 	"github.com/multiformats/go-multihash"
 )
 
-var _ provider.MultihashIterator = (*indexMhIterator)(nil)
+var _ MultihashIterator = (*indexMhIterator)(nil)
 
 type indexMhIterator struct {
 	mhch chan multihash.Multihash
 	err  error
 }
 
-func newIndexMhIterator(ctx context.Context, idx index.IterableIndex) *indexMhIterator {
+// NewMultihashIteratorFromIterableIndex constructs a new MultihashIterator from a CAR index.IterableIndex.
+func NewMultihashIteratorFromIterableIndex(ctx context.Context, idx index.IterableIndex) MultihashIterator {
```

Comment:
Hm, I find this name pretty verbose. How about `IterableIndexIterator`, `CarIndexIterator`, or `CarMultihashIterator`?

I'd also prefer to name the index import `carindex`, to clarify that the parameter is `carindex.IterableIndex`. Otherwise the name "index" is sort of confusing, as it doesn't clarify it means a CAR index.

Reply:
I think I prefer CarMultihashIterator, because it clarifies the input is CAR-specific, and it's still clear that the output is a MultihashIterator.